cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -441,12 +482,14 @@ func (csr *ClusterStateRegistry) updateNodeGroupMetrics() {
 	metrics.UpdateNodeGroupsCount(autoscaled, autoprovisioned)
 }
 
-// IsNodeGroupSafeToScaleUp returns true if node group can be scaled up now.
-func (csr *ClusterStateRegistry) IsNodeGroupSafeToScaleUp(nodeGroup cloudprovider.NodeGroup, now time.Time) bool {
-	if !csr.IsNodeGroupHealthy(nodeGroup.Id()) {
-		return false
+// IsNodeGroupSafeToScaleUp returns information about node group safety to be scaled up now.
+func (csr *ClusterStateRegistry) IsNodeGroupSafeToScaleUp(nodeGroup cloudprovider.NodeGroup, now time.Time) NodeGroupScalingSafety {
+	nodeGroupStatus := csr.GetNodeGroupHealth(nodeGroup.Id())
+	if !nodeGroupStatus.IsHealthy {
+		return NodeGroupScalingSafety{false, nodeGroupStatus.ErrorInfo}
```

Comment:
nit: Using named struct parameters is preferred (no need to modify struct literals whenever a new field is added).

Reply:
Done, thank you!